sdk/keyvault/Azure.Security.KeyVault.Keys/src/RecoverDeletedKeyOperation.cs

```diff
@@ -41,98 +44,49 @@ internal RecoverDeletedKeyOperation(KeyVaultPipeline pipeline, Response<KeyVault
-            }
-
-            return GetRawResponse();
-        }
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default) =>
+            await _operationInternal.UpdateStatusAsync(cancellationToken).ConfigureAwait(false);
 
         /// <inheritdoc />
-        public override ValueTask<Response<KeyVaultKey>> WaitForCompletionAsync(CancellationToken cancellationToken = default) =>
-            this.DefaultWaitForCompletionAsync(s_defaultPollingInterval, cancellationToken);
+        public override async ValueTask<Response<KeyVaultKey>> WaitForCompletionAsync(CancellationToken cancellationToken = default) =>
+            await _operationInternal.WaitForCompletionAsync(cancellationToken).ConfigureAwait(false);
 
         /// <inheritdoc />
-        public override ValueTask<Response<KeyVaultKey>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken) =>
-            this.DefaultWaitForCompletionAsync(pollingInterval, cancellationToken);
+        public override async ValueTask<Response<KeyVaultKey>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken) =>
+            await _operationInternal.WaitForCompletionAsync(pollingInterval, cancellationToken).ConfigureAwait(false);
 
-        private async ValueTask<bool> CheckCompletedAsync(Response response)
-        {
-            switch (response.Status)
-            {
-                case 200:
-                case 403: // Access denied but proof the key was recovered.
-                    return true;
+        async Task<Response> IOperation<KeyVaultKey>.GetResponseAsync(CancellationToken cancellationToken) =>
+            await _pipeline.GetResponseAsync(RequestMethod.Get, cancellationToken, KeyClient.KeysPath, Value.Name, "/", Value.Properties.Version).ConfigureAwait(false);
 
-                case 404:
-                    return false;
+        Response IOperation<KeyVaultKey>.GetResponse(CancellationToken cancellationToken) =>
+            _pipeline.GetResponse(RequestMethod.Get, cancellationToken, KeyClient.KeysPath, Value.Name, "/", Value.Properties.Version);
 
-                default:
-                    throw await _pipeline.Diagnostics.CreateRequestFailedExceptionAsync(response).ConfigureAwait(false);
-            }
-        }
-        private bool CheckCompleted(Response response)
+        OperationState<KeyVaultKey> IOperation<KeyVaultKey>.UpdateState(Response response)
         {
             switch (response.Status)
             {
                 case 200:
                 case 403: // Access denied but proof the key was recovered.
-                    return true;
+                    return OperationState<KeyVaultKey>.Success(Value);
 
                 case 404:
-                    return false;
+                    return default;
 
                 default:
                     throw _pipeline.Diagnostics.CreateRequestFailedException(response);
```

Comment:
Why do we throw instead of returning `Failed`?

Reply:
So we don't mark the operation as completed. I'm doing this so we don't change the original behavior of this operation. Originally, when it throws, it still considers the operation as running.